<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1017</width>
    <height>623</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Owlcat Mono Profiler</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="styleSheet">
       <string notr="true">QSplitter::handle {
    background-color: #AAAAAA;
}</string>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="opaqueResize">
       <bool>true</bool>
      </property>
      <property name="handleWidth">
       <number>5</number>
      </property>
      <property name="childrenCollapsible">
       <bool>false</bool>
      </property>
      <widget class="QWidget" name="layoutWidget">
       <layout class="QVBoxLayout" name="verticalLayout" stretch="1,1,1,0">
        <item>
         <widget class="QwtPlot" name="allocationsGraph">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>100</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>200</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QScrollBar" name="horizontalScrollBar">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QwtPlot" name="sizeGraph">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>100</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>200</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <property name="spacing">
           <number>6</number>
          </property>
          <item>
           <widget class="QCheckBox" name="snapToGC">
            <property name="text">
             <string>Snap to GC</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QTabWidget" name="tabWidget">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <property name="documentMode">
        <bool>false</bool>
       </property>
       <property name="tabsClosable">
        <bool>false</bool>
       </property>
       <widget class="QWidget" name="liveObjectsTab">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="autoFillBackground">
         <bool>false</bool>
        </property>
        <attribute name="title">
         <string>Live Objects</string>
        </attribute>
        <layout class="QGridLayout" name="liveObjectsGrid">
         <item row="1" column="2">
          <widget class="QTableView" name="callstacksList">
           <property name="alternatingRowColors">
            <bool>true</bool>
           </property>
           <property name="selectionBehavior">
            <enum>QAbstractItemView::SelectRows</enum>
           </property>
           <property name="verticalScrollMode">
            <enum>QAbstractItemView::ScrollPerPixel</enum>
           </property>
           <property name="sortingEnabled">
            <bool>true</bool>
           </property>
           <property name="wordWrap">
            <bool>false</bool>
           </property>
           <attribute name="horizontalHeaderStretchLastSection">
            <bool>false</bool>
           </attribute>
           <attribute name="verticalHeaderVisible">
            <bool>false</bool>
           </attribute>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="liveObjectsSizeLabel">
           <property name="text">
            <string>Selected 0Mb of 0Mb</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QTableView" name="liveObjectsList">
           <property name="palette">
            <palette>
             <active>
              <colorrole role="Highlight">
               <brush brushstyle="SolidPattern">
                <color alpha="255">
                 <red>0</red>
                 <green>120</green>
                 <blue>215</blue>
                </color>
               </brush>
              </colorrole>
             </active>
             <inactive>
              <colorrole role="Highlight">
               <brush brushstyle="SolidPattern">
                <color alpha="255">
                 <red>0</red>
                 <green>120</green>
                 <blue>215</blue>
                </color>
               </brush>
              </colorrole>
             </inactive>
             <disabled>
              <colorrole role="Highlight">
               <brush brushstyle="SolidPattern">
                <color alpha="255">
                 <red>0</red>
                 <green>120</green>
                 <blue>215</blue>
                </color>
               </brush>
              </colorrole>
             </disabled>
            </palette>
           </property>
           <property name="alternatingRowColors">
            <bool>true</bool>
           </property>
           <property name="selectionMode">
            <enum>QAbstractItemView::ExtendedSelection</enum>
           </property>
           <property name="selectionBehavior">
            <enum>QAbstractItemView::SelectRows</enum>
           </property>
           <property name="showGrid">
            <bool>false</bool>
           </property>
           <property name="gridStyle">
            <enum>Qt::NoPen</enum>
           </property>
           <property name="sortingEnabled">
            <bool>true</bool>
           </property>
           <property name="wordWrap">
            <bool>false</bool>
           </property>
           <attribute name="horizontalHeaderStretchLastSection">
            <bool>false</bool>
           </attribute>
           <attribute name="verticalHeaderVisible">
            <bool>false</bool>
           </attribute>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLineEdit" name="typesFilter">
           <property name="placeholderText">
            <string>Filter types</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QLineEdit" name="callstackFilter">
           <property name="placeholderText">
            <string>Filter callstacks</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="referencesTab">
        <attribute name="title">
         <string>References</string>
        </attribute>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QTreeView" name="objectReferences">
           <property name="uniformRowHeights">
            <bool>true</bool>
           </property>
           <attribute name="headerVisible">
            <bool>false</bool>
           </attribute>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1017</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="actionStart_Profiling"/>
   <addaction name="actionRunUnityApp"/>
   <addaction name="actionStop_Profiling"/>
   <addaction name="actionPause_App"/>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionStart_Profiling">
   <property name="text">
    <string>Connect to App</string>
   </property>
   <property name="toolTip">
    <string>Connect to running Unity application or Editor</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="actionStop_Profiling">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Stop Profiling</string>
   </property>
   <property name="shortcut">
    <string>Shift+F5</string>
   </property>
  </action>
  <action name="actionRunUnityApp">
   <property name="text">
    <string>Run App</string>
   </property>
   <property name="toolTip">
    <string>Run and profile Unity application</string>
   </property>
  </action>
  <action name="actionPause_App">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Pause App</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QwtPlot</class>
   <extends>QFrame</extends>
   <header>qwt_plot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionStop_Profiling</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onStopProfiling()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalScrollBar</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>onAllocationsScrolled(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>83</x>
     <y>175</y>
    </hint>
    <hint type="destinationlabel">
     <x>438</x>
     <y>25</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onOpenData()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>452</x>
     <y>327</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStart_Profiling</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onStartProfiling()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>201</x>
     <y>48</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>448</x>
     <y>327</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onSaveData()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>448</x>
     <y>327</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRunUnityApp</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>onRunUnityApp()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>448</x>
     <y>327</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>callstacksList</sender>
   <signal>customContextMenuRequested(QPoint)</signal>
   <receiver>MainWindow</receiver>
   <slot>onLiveObjectsCallstacksContextMenuRequested(QPoint)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>704</x>
     <y>425</y>
    </hint>
    <hint type="destinationlabel">
     <x>756</x>
     <y>574</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPause_App</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>onPauseApp(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>508</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>typesFilter</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>onTypeFilterChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>203</x>
     <y>350</y>
    </hint>
    <hint type="destinationlabel">
     <x>4</x>
     <y>532</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>callstackFilter</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>onCallstackFilterChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>604</x>
     <y>356</y>
    </hint>
    <hint type="destinationlabel">
     <x>1011</x>
     <y>432</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>objectReferences</sender>
   <signal>expanded(QModelIndex)</signal>
   <receiver>MainWindow</receiver>
   <slot>onReferenceExpanded(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>416</x>
     <y>421</y>
    </hint>
    <hint type="destinationlabel">
     <x>598</x>
     <y>599</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onAllocationsScrolled(int)</slot>
  <slot>onOpenData()</slot>
  <slot>onStartProfiling()</slot>
  <slot>onStopProfiling()</slot>
  <slot>onSaveData()</slot>
  <slot>onRunUnityApp()</slot>
  <slot>onLiveObjectsCallstacksContextMenuRequested(QPoint)</slot>
  <slot>onPauseApp(bool)</slot>
  <slot>onTypeFilterChanged(QString)</slot>
  <slot>onCallstackFilterChanged(QString)</slot>
  <slot>onReferenceExpanded(QModelIndex)</slot>
 </slots>
</ui>
